src/utils/crypto.ts

```diff
@@ -106,3 +107,49 @@ export const getSigningJwk = (publicKey: RSAKey | KJUR.crypto.ECDSA): JWK => ({
   ...JWK.parse(KEYUTIL.getJWKFromKey(publicKey)),
   use: "sig",
 });
+
+/**
+ * This function takes two {@link PublicKey} and executes on them
+ * a deep comparison, checking if they have the same kty, and,
+ * in case they both are RSA keys, wether they use the same alg and
+ * have the same e and n while, in case they both are EC keys, wether they
+ * are defined on the same crv, and have the same x and y
+ * @param key1 The first key
+ * @param key2 The second key
+ * @returns true if the keys are deep equal, false otherwise
+ */
+export const deepCompareKeys = (key1: PublicKey, key2: PublicKey) => {
+  if (key1.kty === "EC" && key2.kty === "EC") {
+    return key1.crv === key2.crv && key1.x === key2.x && key1.y === key2.y;
+  }
+  if (key1.kty === "RSA" && key2.kty === "RSA") {
+    return key1.alg === key2.alg && key1.e === key2.e && key1.n === key2.n;
+  }
+
+  return false;
+};
```

Comment:
This comparison is usually not recommended. The best way to verify that two keys are the same is to calculate the thumbprint of both and compare it. The methods for calculating the thumbprint are already present here.